src/components/vuestic-components/va-form/VaForm.vue

```diff
@@ -0,0 +1,163 @@
+<template>
+  <conponent
+    class="va-form"
+    :is="tag"
+    v-on="$listeners"
+    @submit="submit($event)"
+    @reset="reset()"
+    @validation="validation()"
+    @resetValidation="resetValidation()"
+  >
+    <slot />
+  </conponent>
+</template>
+
+<script>
+import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+
+// TODO: need to remove this line after component-form methods implementation (focus, clear, etc.)
+const availableFormComponents = ['VaInput', 'VaColorInput', 'VaDatePicker', 'VaCheckbox', 'VaRadio', 'VaSelect', 'VaFileUpload']
+const getAllChildren = (vm, children = []) => {
+  vm.$children.forEach((child) => {
+    children.push(child)
+    child.$children.length > 0 && getAllChildren(child, children)
+  })
+  return children
+}
+
+export default {
+  name: 'VaForm',
+  mixins: [ContextPluginMixin],
+  props: {
+    lazyValidation: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'lazyValidation', false)
+      },
+    },
+    autofocus: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'autofocus', false)
+      },
+    },
+    tag: {
+      type: String,
+      default () {
+        return getContextPropValue(this, 'tag', 'div')
+      },
+    },
+  },
+  data () {
+    return {
+      valid: true,
+      hasResetButton: false, // Need for form as div tag
+      hasSubmitButton: false, // Need for form as div tag
+    }
+  },
+  mounted () {
+    this.$el.addEventListener('focusin', this.focus)
+
+    if (this.autofocus) {
+      const firstFormChild = getAllChildren(this).find((child) => availableFormComponents.includes(child.$options.name))
+
+      if (firstFormChild) {
+        firstFormChild.$el.focus()
+      }
+    }
+
+    if (this.tag !== 'form') { // component dont have tag form, we need set listeners manually
+      const resetButton = this.$el.querySelector('button[type=reset]')
+
+      if (resetButton) {
+        resetButton.addEventListener('click', this.reset)
+
+        this.hasResetButton = true
+      }
+
+      const submitButton = this.$el.querySelector('button[type=submit]')
+
+      if (submitButton) {
+        submitButton.addEventListener('click', this.submit)
```

Comment:
I'm not sure we need to listen to the click event on the submit button. Clicking on the submit button will trigger form submit event by default:
```
    <form @submit="doSmth">
      <button type="submit">
        go
      </button>
    </form>
```

Reply:
We can support form inside form cases. Low-level form can be as a div tag

Reply:
@jean-moldovan it's likely we're going to choose the strategy you're suggesting.